src/result_response.cpp

```diff
@@ -195,7 +210,8 @@ class DataTypeDecoder {
 
 private:
   char* buffer_;
-  DataType::Ptr data_type_cache_[CASS_VALUE_TYPE_LAST_ENTRY];
+  sparsehash::dense_hash_map<uint16_t, DataType::Ptr> data_type_cache_;
```

Comment:
I'd like to move this back to using an array now that the potential values are more dense.